addons/mail_group/tests/test_mail_group.py

```diff
@@ -78,6 +78,50 @@ def test_find_member(self):
         member = self.test_group._find_member(email, partner_2.id)
         self.assertFalse(member, 'Should not return any member because the only one with the same email has a different partner')
 
+    def test_find_member_for_alias(self):
+        """Test the matching of a mail_group.members, when 2 users have the same partner email, and
+        that the first user was subscribed."""
+        email = 'Alias <test_find_member_for_alias@odoo.com>'
+
+        alias = self.env['mail.alias'].create({
+            'alias_name': 'test_find_member_for_alias',
+            'alias_model_id': self.env.ref('base.model_res_partner').id,
+            'alias_contact': 'followers',
+        })
+
+        user = self.env['res.users'].create({
+            'login': 'login_1',
+            'name': 'login_1',
+            'email': email,
+        })
+
+        _ = self.env['res.users'].create({
+            'login': 'login_2',
+            'name': 'login_2',
+            'email': email,
+        })
+        self.test_group.access_mode == 'followers'
+
+        member = self.env['mail.group.member'].create({
+            'partner_id': user.partner_id.id,
+            'mail_group_id': self.test_group.id,
+        })
+        assert member.email == email
+
+        msg_dict = {
+            'author_id': self.env['mail.thread']._mail_find_user_for_gateway(email).id,
```

Comment:
can you add a sanity check that
```
self.env['mail.thread']._mail_find_user_for_gateway(email) == user
```
?

Reply:
`author_id` is a partner not a user isn't it ?

Reply:
not it is a user :/

Reply:
Author is a partner... message_dict is normally the result of parsing, which means values suitable for a mail.message. Are you sure those are users ?